hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java

```diff
@@ -132,6 +137,7 @@ public QueryWorkflowImpl(
         this.dispatcher = requireNonNull(dispatcher);
         this.feeAccumulator = requireNonNull(feeAccumulator);
         this.queryContext = requireNonNull(queryContext);
+        this.queryParser = requireNonNull(queryParser);
```

Comment:
Previously parsers had to be in the session context. Is that not required anymore, i.e. are they thread-safe?

Reply:
Good question, I'm not sure. I replaced the call to parse the query with an object field here because the tests have mocked a query parser as if this class already has one, so I figured something had not been ported over.

Reply:
@jasperpotts determined in our modularization meeting this morning that the parsers are stateless, so they are threadsafe. There is a caveat: the data inputs and outputs won't necessarily be threadsafe.